src/ripple/app/misc/SHAMapStoreImp.h

```diff
@@ -135,12 +135,6 @@ class SHAMapStoreImp : public Stoppable, public SHAMapStore
         NodeStore::Scheduler& scheduler,
         beast::Journal journal);
 
-    ~SHAMapStoreImp()
-    {
-        if (thread_.joinable())
-            thread_.join();
-    }
```

Comment:
If I understand correctly, if `onStop` isn't called, then the `thread_` destructor will throw when `SHAMapStoreImp` is destroyed. I could see where that is intentional if you want to ensure that `onStop` is always called, but I wanted to confirm it's intentional.

Reply:
Yes, it's intentional. `std::thread` is the model for the new design. This guarantees that the stop sequence for `ApplicationImp` is the same every time, because `ApplicationImp::stop` _must_ be called.